pulseapi/utility/contributions_data.py

```diff
@@ -64,16 +92,34 @@ def extract_data(json_data):
 
 def create_events_csv():
     rows = []
+    repo_error = []
+
+    session = requests.Session()
+    adapter = requests.adapters.HTTPAdapter(max_retries=3)
+    session.mount('https://', adapter)
 
     for repo in repos:
         print(f'Fetching Activity for: {repo}')
         for page in range(1, 11):
-            r = requests.get(f'https://api.github.com/repos/{repo}/events?page={page}&access_token={token}')
-            extracted_data = extract_data(r.json())
-
-            for event in extracted_data:
-                row = [event.id, event.created_at, event.type, event.action, event.contributor, event.repo]
-                rows.append(row)
+            try:
+                r = session.get(f'https://api.github.com/repos/{repo}/events?page={page}&access_token={token}')
+            except MaxRetryError as err:
```

Comment:
nice!

Is it possible to check for rate limit errors? (427 I think?)

Reply:
Github API return of 403 when you reach the limit (https://developer.github.com/v3/#rate-limiting). I guess we could check the header for the amount left ('X-RateLimit-Remaining'). and stop if it's at zero. Do you want me to do that?